docs: add jcloud executor availability parameters (#5624)
jina-ai/jina: docs/concepts/jcloud/yaml-spec.md

After JCloud deployment using the autoscaling configuration, the Flow serving part is just the same; the only difference you may notice is it takes a few extra seconds to handle the initial requests since it needs to scale the deployments behind the scenes. Let JCloud handle the scaling from now on, and you should only worry about the code!

## Configure Availability Tolerance

> In case of any unlikely event that would cause the service disruption of executors, JCloud allows users to specify a tolerance level in terms of number of replicas.

Reviewer: ```suggestion
If service issues cause disruption of Executors, JCloud lets you specify a tolerance level for number of replicas that stay up or go down.
```

> The JCloud parameters minAvailable and maxUnavailable ensures the absense of some replicas of an executor to be with in the limits of toleration.

Reviewer: ```suggestion
The JCloud parameters `minAvailable` and `maxUnavailable` ensure that Executors will stay up even if a certain number of replicas go down.
```
Please spell-check in future :)

I did quite a significant rewrite. Is it still accurate?

| Name           | Default |                                     Allowed                                     | Description                                                      |
|:---------------|:-------:|:-------------------------------------------------------------------------------:|:-----------------------------------------------------------------|

>  | minAvailable   |   NA    | lesser than [replicas](https://docs.jina.ai/concepts/flow/scale-out/#scale-out) | the minimum number of replicas available during the disruption   |
>  | maxUnavailable |   NA    | lesser than [replicas](https://docs.jina.ai/concepts/flow/scale-out/#scale-out) | the maximum number of replicas unavailable during the disruption |

Reviewer: ```suggestion
 | `minAvailable`   |   N/A    | Lower than number of [replicas](https://docs.jina.ai/concepts/flow/scale-out/#scale-out) | Minimum number of replicas available during disruption   |
 | `maxUnavailable` |   N/A    | Lower than numbers of [replicas](https://docs.jina.ai/concepts/flow/scale-out/#scale-out) | Maximum number of replicas unavailable during disruption |
```

```{code-block} yaml
---
emphasize-lines: 5-6
---
jtype: Flow
executors:
  - uses: jinahub+docker://Sentencizer
    replicas: 5
    jcloud:
      minAvailable: 2
```

> > In case of unlike disruption, it is ensured that at least 2 replicas would still be available. 3 replicas may be unavailable.

Reviewer: ```suggestion
> In case of unlike disruption, ensure at least two replicas will still be available, while 3 may be down.
```
- What does 'unlike' mean? unlikely?

```{code-block} yaml
---
emphasize-lines: 5-6
---
jtype: Flow
executors:
  - uses: jinahub+docker://Sentencizer
    replicas: 5
    jcloud:
      maxUnavailable: 2
```

> > In case of unlike disruption, it is ensured that at maximum 2 replicas would not be available. At least 3 replicas would still be available.

Reviewer: ```suggestion
> In case of unlike disruption, ensure that if a maximum of 2 replicas are down, at least 3 replicas will still be available.
```
Again, unlike means...?

> ## Configure gateway

Reviewer: ```suggestion
## Configure Gateway
```

> After JCloud deployment using the autoscaling configuration, the Flow serving part is just the same; the only difference you may notice is it takes a few extra seconds to handle the initial requests since it needs to scale the deployments behind the scenes. Let JCloud handle the scaling from now on, and you should only worry about the code!

Reviewer: ```suggestion
After JCloud deployment using the autoscaling configuration, the Flow serving part is just the same: the only difference you may notice is it takes a few extra seconds to handle the initial requests since it needs to scale the deployments behind the scenes. Let JCloud handle the scaling from now on, and you can deal with the code!
```

> ## Configure Availability Tolerance

Reviewer: ```suggestion
## Configure availability tolerance
```

If service issues cause disruption of Executors, JCloud lets you specify a tolerance level for number of replicas that stay up or go down.

The JCloud parameters `minAvailable` and `maxUnavailable` ensure that Executors will stay up even if a certain number of replicas go down.

| Name             | Default |                                          Allowed                                          | Description                                              |
|:-----------------|:-------:|:-----------------------------------------------------------------------------------------:|:---------------------------------------------------------|
 | `minAvailable`   |   N/A   | Lower than number of [replicas](https://docs.jina.ai/concepts/flow/scale-out/#scale-out)  | Minimum number of replicas available during disruption   |
| `maxUnavailable` |   N/A   | Lower than numbers of [replicas](https://docs.jina.ai/concepts/flow/scale-out/#scale-out) | Maximum number of replicas unavailable during disruption |

```{code-block} yaml
---
emphasize-lines: 5-6
---
jtype: Flow
executors:

>   - uses: jinahub+docker://Sentencizer

Reviewer: Should this be `jinaai://...`?

    replicas: 5
    jcloud:
      minAvailable: 2
```

> > In case of unlike disruption, ensure at least two replicas will still be available, while 3 may be down.

Reviewer: ```suggestion
> In case of disruption, ensure at least two replicas will still be available, while three may be down.
```

```{code-block} yaml
---
emphasize-lines: 5-6
---
jtype: Flow
executors:
  - uses: jinahub+docker://Sentencizer
    replicas: 5
    jcloud:
      maxUnavailable: 2
```

> > In case of unlike disruption, ensure that if a maximum of 2 replicas are down, at least 3 replicas will still be available.

Reviewer: ```suggestion
> In case of disruption, ensure that if a maximum of two replicas are down, at least three replicas will still be available.
```

---
jtype: Flow
executors:
  - uses: jinaai+docker://Sentencizer
    replicas: 5
    jcloud:
      minAvailable: 2
```
> In case of disruption, ensure at least two replicas will still be available, while three may be down.

```{code-block} yaml
---
emphasize-lines: 5-6
---
jtype: Flow
executors:

>   - uses: jinaai+docker://Sentencizer

Reviewer: Is this missing the username portion?
![image](https://user-images.githubusercontent.com/4182659/215782787-45d56346-e115-49d9-bcd3-e5c9e6b12419.png)

https://cloud.jina.ai/executor/c6focg47
